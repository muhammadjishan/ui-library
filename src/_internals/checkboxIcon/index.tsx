import React from 'react'

import { color } from '../../_utils/branding'
import { CheckIcon } from '../../icon/checkIcon'
import { CircleIcon } from '../../icon/circleIcon'
import { Loader } from '../../loader'

export interface CheckboxIconProps {
  readonly isChecked?: boolean
  readonly isLoading?: boolean
  readonly isDisabled?: boolean
}

export const CheckboxIcon = ({
  isChecked = false,
  isLoading = false,
  isDisabled = false,
}: CheckboxIconProps) => {
  if (isLoading) {
    return <Loader size={24} inline />
  }
  if (isChecked) {
    return (
      <CheckIcon
        backgroundColor={color.blue}
        iconColor={color.white}
        isDisabled={isDisabled}
        thin
      />
    )
  }
  return <CircleIcon iconColor={color.blue} isDisabled={isDisabled} />
}

export default CheckboxIcon
