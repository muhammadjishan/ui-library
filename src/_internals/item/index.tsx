import styled from 'styled-components'

import { color, space } from '../../_utils/branding'
import { Item } from './Item'

const StyledItem = styled(Item)`
  & {
    position: relative;
    display: flex;
    padding: 0;
    padding-top: ${space.l};
    padding-bottom: ${space.l};
    align-items: center;
    flex: 1;
    border: 0;
    background: none;
  }

  &.kirk-item--clickable:hover {
    cursor: pointer;
    background: ${color.lightGray};
  }

  &.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
    background: none;
  }

  &:disabled {
    cursor: default;
    background-color: ${color.defaultBackground};
  }

  & {
    background: none;
    text-decoration: none;
    user-select: none;
    -webkit-tap-highlight-color: ${color.tapHighlight};
    -webkit-touch-callout: none;
  }
  /* Button tag fixes */
  button& {
    cursor: pointer;
    text-align: left;
    width: 100%;
    font-family: inherit;
    background-color: transparent;
    -webkit-tap-highlight-color: ${color.tapHighlight};
    -webkit-touch-callout: none;
  }

  button:focus:not(.focus-visible)& {
    outline: none;
  }

  /* Text areas */
  & .kirk-item-leftText {
    flex: 1;
  }

  & .kirk-item-rightText {
    margin-left: ${space.l};
    text-align: right;
  }

  /* Text */
  & .kirk-text {
    display: block;
  }

  & .kirk-item-title--withBody {
    margin-bottom: ${space.s};
  }

  & .kirk-item--strikethrough {
    text-decoration: line-through;
  }

  & .kirk-item-title--withButtonAddon {
    position: relative;
  }

  /* Addons */
  & .kirk-item-leftAddon,
  & .kirk-item-rightAddon {
    display: inline-flex;
    min-width: 24px;
    align-items: center;
  }

  & .kirk-item-leftAddon {
    margin-right: ${space.l};
    /* //icons */
    color: ${color.mindNightGreen};
  }

  & .kirk-item-rightAddon {
    margin-left: ${space.l};
  }

  /* Highlighted state */
  &.kirk-item--highlighted .kirk-text-title {
    color: ${color.blue};
  }

  /* Button info */
  & .kirk-item-title--withButtonAddon button {
    position: absolute;
    width: 24px;
    height: 24px;
    margin-left: ${space.s};
    bottom: 0;
  }
`

export { ItemStatus } from './Item'
export { StyledItem as Item }
export default StyledItem
