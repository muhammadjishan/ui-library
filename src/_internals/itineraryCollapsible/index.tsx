import styled from 'styled-components'

import { color, componentSizes, font, radius, space, transition } from '../../_utils/branding'
import { ItineraryCollapsible } from './ItineraryCollapsible'

const minHeight = '40px'
const smallBulletPositionDifference = `${Math.ceil(
  (parseInt(componentSizes.bulletSize, 10) - parseInt(componentSizes.bulletSizeSmall, 10)) / 2,
)}px`

const StyledItineraryCollapsible = styled(ItineraryCollapsible)`
  & {
    position: relative;
    display: block;
    list-style-type: none;
    cursor: pointer;
    min-height: ${minHeight};
  }

  &.kirk-itineraryCollapsible--extended .kirk-itineraryCollapsible-collapsed,
  &.kirk-itineraryCollapsible--collapsed .kirk-itineraryCollapsible-extended {
    opacity: 0;
    height: ${minHeight} !important; /* must take priority over the real height specified inline */
    overflow: hidden;
  }

  .kirk-itineraryCollapsible-collapsed,
  .kirk-itineraryCollapsible-extended {
    opacity: 1;
    transition: height ${transition.duration.base} ease-in-out,
      opacity ${transition.duration.base} linear;
  }

  & .kirk-itineraryCollapsible-collapsed {
    display: flex;
    position: absolute;
    top: 0;
    left: 0;
    padding: ${space.m} 0;
    height: ${minHeight};
    background-color: ${color.white};
  }

  & .kirk-itineraryCollapsible-collapsed::before,
  & .kirk-itineraryCollapsible-collapsed::after {
    content: '';
    position: absolute;
    top: 0;
    left: 2px;
    width: ${componentSizes.roadWidth};
    height: 4px;
    background-color: ${color.midnightGreen};
    border-radius: ${radius.s};
  }

  & .kirk-itineraryCollapsible-collapsed::before {
    border-top-left-radius: 0;
    border-top-right-radius: 0;
  }

  & .kirk-itineraryCollapsible-collapsed::after {
    top: 30px;
    height: 16px;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
  }

  & .kirk-itineraryCollapsible-collapsed .kirk-bullet {
    position: relative;
    top: 6px;
    left: 0;
  }

  & .kirk-itineraryCollapsible-extended .kirk-bullet {
    top: ${smallBulletPositionDifference};
  }

  & .kirk-itineraryCollapsible-collapsed .kirk-text {
    padding-left: ${space.m};
    height: ${font.m.lineHeight};
  }

  & .kirk-itineraryLocation-wrapper .kirk-itineraryLocation-road {
    height: calc(
      100% - ${componentSizes.bulletSizeSmall} + 4px
    ); /* +4px to better hide behind small Bullets */
  }

  & .kirk-itineraryLocation-roadContainer {
    margin-top: 0;
  }

  & .kirk-itineraryLocation-wrapper {
    padding: 0;
  }
`
export { ItineraryCollapsibleProps } from './ItineraryCollapsible'
export { StyledItineraryCollapsible as ItineraryCollapsible }
export default StyledItineraryCollapsible
