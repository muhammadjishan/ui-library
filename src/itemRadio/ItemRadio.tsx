import React, { Component, Fragment } from 'react'
import cc from 'classcat'

import { Item } from '../_internals/item'
import { OnChangeParameters } from '../_internals/onChange'
import { RadioIcon } from '../_internals/radioIcon'
import { A11yProps, pickA11yProps } from '../_utils/interfaces'
import { TextDisplayType } from '../text'

export enum ItemRadioStatus {
  DEFAULT = 'default',
  LOADING = 'loading',
}

export interface ItemRadioProps extends A11yProps {
  readonly className?: string
  readonly name: string
  readonly value: string | number
  readonly leftAddon?: React.ReactNode
  readonly labelTitle?: string
  readonly label: string
  readonly data?: string
  readonly dataInfo?: string
  readonly checked?: boolean
  readonly disabled?: boolean
  readonly chevron?: boolean
  readonly highlighted?: boolean
  readonly onChange?: (obj: OnChangeParameters) => void
  readonly onClick?: (obj: OnChangeParameters) => void
  readonly status?: ItemRadioStatus
  readonly key?: string | number
}

interface ItemRadioState {
  readonly focus: boolean
}

export class ItemRadio extends Component<ItemRadioProps> {
  static defaultProps: Partial<ItemRadioProps> = {
    onChange() {},
    onClick() {},
    checked: false,
  }

  state: ItemRadioState = {
    focus: false,
  }

  onChange = () => {
    const { name, value } = this.props
    this.props.onChange({ name, value })
  }

  onClick = () => {
    const { name, value } = this.props
    this.props.onClick({ name, value })
  }

  onFocus = () => {
    this.setState({ focus: true })
  }

  onBlur = () => {
    this.setState({ focus: false })
  }

  render() {
    const {
      label,
      name,
      value,
      data,
      className,
      labelTitle,
      dataInfo,
      checked,
      disabled,
      status,
      chevron,
      highlighted,
      leftAddon,
    } = this.props
    const a11yAttrs = pickA11yProps<ItemRadioProps>(this.props)
    const isLoading = status === ItemRadioStatus.LOADING
    const radio = (
      <Fragment>
        <input
          type="radio"
          name={name}
          value={value}
          checked={checked}
          onChange={this.onChange}
          onClick={this.onClick}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          disabled={disabled || isLoading}
        />
        {(!chevron || isLoading) && (
          <RadioIcon isChecked={checked} isLoading={isLoading} isDisabled={disabled} />
        )}
      </Fragment>
    )

    return (
      <Fragment>
        <Item
          className={cc(['kirk-item-radio', className, { focus: this.state.focus }])}
          leftTitle={labelTitle}
          leftBody={label}
          rightTitle={data}
          rightTitleDisplay={TextDisplayType.SUBHEADERSTRONG}
          rightBody={dataInfo}
          /* No a11y issue here
            - The input is well wrapped with the label
            - The linter can't access the complex components implementation
          */
          // eslint-disable-next-line jsx-a11y/label-has-associated-control
          tag={<label />}
          leftAddon={leftAddon}
          rightAddon={radio}
          chevron={chevron && !isLoading}
          highlighted={highlighted}
          isClickable={!disabled}
          disabled={disabled}
          {...a11yAttrs}
        />
      </Fragment>
    )
  }
}
