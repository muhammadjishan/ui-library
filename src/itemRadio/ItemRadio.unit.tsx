import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

import { Item } from '../_internals/item'
import { ItemRadio, ItemRadioProps, ItemRadioStatus } from './ItemRadio'

describe('ItemRadio', () => {
  const defaultProps: ItemRadioProps = {
    label: 'Label',
    name: 'name',
    value: 0,
    className: 'custom-class-name',
    labelTitle: 'Label title',
    data: 'Data',
    dataInfo: 'Data info',
    checked: false,
    disabled: false,
    onChange() {},
    onClick() {},
    status: ItemRadioStatus.DEFAULT,
    key: 0,
  }

  it('Should use the Item component', () => {
    const itemRadio = shallow(<ItemRadio {...defaultProps} />)
    expect(itemRadio.find(Item).exists()).toBe(true)
  })
  it('Should forward its props to the Item component', () => {
    const itemRadio = renderer.create(<ItemRadio {...defaultProps} />).toJSON()
    expect(itemRadio).toMatchSnapshot()
  })
  it('Should display a CircleIcon with an innerDisc when the input is checked', () => {
    const itemRadio = renderer.create(<ItemRadio {...defaultProps} checked />).toJSON()
    expect(itemRadio).toMatchSnapshot()
  })
  it('Should display a Loader when the component is in loading status', () => {
    const itemRadio = renderer
      .create(<ItemRadio {...defaultProps} status={ItemRadioStatus.LOADING} />)
      .toJSON()
    expect(itemRadio).toMatchSnapshot()
  })
  it('Should handle disabled prop', () => {
    const itemRadio = shallow(<ItemRadio {...defaultProps} />)
    expect(itemRadio.find(Item).prop('isClickable')).toBeTruthy()
    expect(itemRadio.find(Item).prop('disabled')).toBeFalsy()

    itemRadio.setProps({ disabled: true })
    expect(itemRadio.find(Item).prop('isClickable')).toBeFalsy()
    expect(itemRadio.find(Item).prop('disabled')).toBeTruthy()
  })

  describe('a11y', () => {
    it('Should have a label tag as wrapper to associate content text to input radio', () => {
      const itemRadio = mount(<ItemRadio {...defaultProps} />)
      const labelTag = itemRadio.find('label input[type="radio"]')
      expect(labelTag.exists()).toBe(true)
    })
  })

  describe('onChange', () => {
    it('Should bind the onChange callback of the input', () => {
      const itemRadio = mount(<ItemRadio {...defaultProps} />)
      expect(itemRadio.find('input').prop('onChange')).toEqual(itemRadio.instance().onChange)
    })

    it('Should call the onChange prop with name and value when the input changes', () => {
      const onChangeMock = jest.fn()
      const itemRadio = shallow(<ItemRadio {...defaultProps} onChange={onChangeMock} />)
      itemRadio.instance().onChange()
      expect(onChangeMock).toHaveBeenCalledWith({
        name: defaultProps.name,
        value: defaultProps.value,
      })
    })
  })

  describe('onClick', () => {
    it('Should bind the onClick callback of the input', () => {
      const itemRadio = mount(<ItemRadio {...defaultProps} />)
      expect(itemRadio.find('input').prop('onClick')).toEqual(itemRadio.instance().onClick)
    })

    it('Should call the onClick prop with name and value when the input changes', () => {
      const onClickMock = jest.fn()
      const itemRadio = shallow(<ItemRadio {...defaultProps} onClick={onClickMock} />)
      itemRadio.instance().onClick()
      expect(onClickMock).toHaveBeenCalledWith({
        name: defaultProps.name,
        value: defaultProps.value,
      })
    })
  })
})
