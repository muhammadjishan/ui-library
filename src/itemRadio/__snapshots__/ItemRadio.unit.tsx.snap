// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRadio Should display a CircleIcon with an innerDisc when the input is checked 1`] = `
.c3 {
  fill: #00AFF5;
}

.c3.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c3.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c3 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1 {
  margin: 0;
  font-weight: 400;
}

.c1.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-body,
.c1.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c1.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c1.kirk-text-subheader,
.c1.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-bodyStrong,
.c1.kirk-text-subheaderStrong,
.c1.kirk-text-titleStrong {
  font-weight: 500;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
}

.c0.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c0.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c0:disabled {
  cursor: default;
}

a.c0 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c0 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c0 {
  outline: none;
}

.c0 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c0 .kirk-text {
  display: block;
}

.c0 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c0 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c0 .kirk-item-leftAddon,
.c0 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c0 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c0.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c0 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c2.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c2 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c2.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c2 circle.outer {
  stroke: #00AFF5;
}

.c2 circle.inner {
  stroke-width: 0;
  fill: #00AFF5;
}

.c2.thin circle {
  stroke-width: 3;
}

.c2.absolute {
  position: absolute;
}

<label
  className="kirk-item kirk-item--clickable kirk-item-radio custom-class-name c0"
>
  <span
    className="kirk-item-leftText"
  >
    <span
      className="kirk-text kirk-text-title c1"
    >
      Label title
    </span>
    <span
      className="kirk-text kirk-text-body kirk-item-body c1"
    >
      Label
    </span>
  </span>
  <span
    className="kirk-item-rightText"
  >
    <div
      className="kirk-text kirk-text-subheaderStrong kirk-item-title c1"
    >
      Data
    </div>
    <div
      className="kirk-text kirk-text-body kirk-item-body c1"
    >
      Data info
    </div>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <input
      checked={true}
      disabled={false}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={0}
    />
    <svg
      aria-hidden={true}
      className="kirk-icon kirk-icon-circle c2 c3"
      height={24}
      viewBox="0 0 66 66"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <circle
        className="outer"
        cx="33"
        cy="33"
        fill="none"
        r="30"
      />
      <circle
        className="inner"
        cx="33"
        cy="33"
        r="18"
      />
    </svg>
  </span>
</label>
`;

exports[`ItemRadio Should display a Loader when the component is in loading status 1`] = `
.c4 {
  fill: #5DD167;
}

.c4.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c4.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c4 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1 {
  margin: 0;
  font-weight: 400;
}

.c1.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-body,
.c1.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c1.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c1.kirk-text-subheader,
.c1.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-bodyStrong,
.c1.kirk-text-subheaderStrong,
.c1.kirk-text-titleStrong {
  font-weight: 500;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
}

.c0.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c0.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c0:disabled {
  cursor: default;
}

a.c0 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c0 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c0 {
  outline: none;
}

.c0 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c0 .kirk-text {
  display: block;
}

.c0 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c0 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c0 .kirk-item-leftAddon,
.c0 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c0 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c0.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c0 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c3.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c3 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c3.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c3 circle.outer {
  stroke: #5DD167;
}

.c3 circle.inner {
  stroke-width: 0;
  fill: #5DD167;
}

.c3.thin circle {
  stroke-width: 3;
}

.c3.absolute {
  position: absolute;
}

.c2.kirk-loader--inline {
  display: inline-block;
}

.c2 .visually-hidden {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  white-space: nowrap;
}

.c2.kirk-loader--fullScreen {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgba(255,255,255,0.85);
  height: 100vh;
  width: 100vw;
  z-index: 4;
}

.c2.kirk-loader--block {
  display: block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 0;
}

.c2 .kirk-loader--done {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5DD167;
  color: #FFF;
  border-radius: 50%;
}

<label
  className="kirk-item kirk-item--clickable kirk-item-radio custom-class-name c0"
>
  <span
    className="kirk-item-leftText"
  >
    <span
      className="kirk-text kirk-text-title c1"
    >
      Label title
    </span>
    <span
      className="kirk-text kirk-text-body kirk-item-body c1"
    >
      Label
    </span>
  </span>
  <span
    className="kirk-item-rightText"
  >
    <div
      className="kirk-text kirk-text-subheaderStrong kirk-item-title c1"
    >
      Data
    </div>
    <div
      className="kirk-text kirk-text-body kirk-item-body c1"
    >
      Data info
    </div>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <input
      checked={false}
      disabled={true}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={0}
    />
    <div
      aria-valuetext="Loading"
      className="c2 kirk-loader--inline"
      role="progressbar"
    >
      <div
        className=""
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="kirk-icon kirk-icon-circle spinning c3 c4"
          height={24}
          viewBox="0 0 66 66"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <circle
            className="outer"
            cx="33"
            cy="33"
            fill="none"
            r="30"
          />
        </svg>
      </div>
    </div>
    <div
      className="visually-hidden"
      role="status"
    >
      Loading
    </div>
  </span>
</label>
`;

exports[`ItemRadio Should forward its props to the Item component 1`] = `
.c3 {
  fill: #00AFF5;
}

.c3.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c3.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c3 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1 {
  margin: 0;
  font-weight: 400;
}

.c1.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-body,
.c1.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c1.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c1.kirk-text-subheader,
.c1.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-bodyStrong,
.c1.kirk-text-subheaderStrong,
.c1.kirk-text-titleStrong {
  font-weight: 500;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
}

.c0.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c0.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c0:disabled {
  cursor: default;
}

a.c0 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c0 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c0 {
  outline: none;
}

.c0 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c0 .kirk-text {
  display: block;
}

.c0 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c0 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c0 .kirk-item-leftAddon,
.c0 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c0 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c0.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c0 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c2.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c2 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c2.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c2 circle.outer {
  stroke: #00AFF5;
}

.c2 circle.inner {
  stroke-width: 0;
  fill: #00AFF5;
}

.c2.thin circle {
  stroke-width: 3;
}

.c2.absolute {
  position: absolute;
}

<label
  className="kirk-item kirk-item--clickable kirk-item-radio custom-class-name c0"
>
  <span
    className="kirk-item-leftText"
  >
    <span
      className="kirk-text kirk-text-title c1"
    >
      Label title
    </span>
    <span
      className="kirk-text kirk-text-body kirk-item-body c1"
    >
      Label
    </span>
  </span>
  <span
    className="kirk-item-rightText"
  >
    <div
      className="kirk-text kirk-text-subheaderStrong kirk-item-title c1"
    >
      Data
    </div>
    <div
      className="kirk-text kirk-text-body kirk-item-body c1"
    >
      Data info
    </div>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <input
      checked={false}
      disabled={false}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={0}
    />
    <svg
      aria-hidden={true}
      className="kirk-icon kirk-icon-circle c2 c3"
      height={24}
      viewBox="0 0 66 66"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <circle
        className="outer"
        cx="33"
        cy="33"
        fill="none"
        r="30"
      />
    </svg>
  </span>
</label>
`;
